### PR TITLE
feat: Ability to keep the dropdown open always (#241) ✨

### PR DESCRIPTION
**Pull request by moonjy1993**
## What does it do?

This adds a new prop: showDropdownAlways, and this disables the toggling of showDropdown when the user clicks search box.

## Fixes # (issue)

This closes issue #239. 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] Updated documentation (if applicable)
- [x] Added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] My changes generate no new warnings
- [ ] Added myself to contributors table
- [ ] I have updated `docs/bundle.js` file (ran `yarn build:docs` and checked in the new `bundle.js` file)

### CODE REVIEW

**Comment by mrchief** [comment]
@moonjy1993 Thanks for submitting this. There are a few things we need to do before this PR can be merged:

 - Add unit tests for newly added functionality
 - Update Readme for the new prop
 - Update the PR description according to the checklist.

**Comment by ellinge** [comment]
Probably should think about bundling this as well.
Change showDropdown to ’onfocus’ (default) | ’initial’ | ’always’ or something similar instead?

**Comment by moonjy1993** [comment]
@ellinge 
but what happens if the user who's already using showDropdown to true in their code?

**Comment by mrchief** [comment]
> Probably should think about bundling this as well.
> Change showDropdown to ’onfocus’ (default) | ’initial’ | ’always’ or something similar instead?

Yes, but we can make that part of #242

> but what happens if the user who's already using showDropdown to true in their code?

They would need a code update when they migrate to the new version, which is why it's a separate release.

**Comment by moonjy1993** [comment]
@mrchief  okay. so I have updated the index.test.js and readMe files. will include showDropdownAlways for now, since you said the logic building will happen in other PR.

**Comment by ellinge** [comment]
The typescript typings (.d.ts) needs an update as well. It’s a very newly added addition so there’s no checklist or similar which ensures this yet.

**Comment by moonjy1993** [comment]
@ellinge  okay. just modified d.ts file.

**Comment by mrchief** [comment]
> The typescript typings (.d.ts) needs an update as well. It’s a very newly added addition so there’s no checklist or similar which ensures this yet.

Gotta remember this myself! :)

File: src/index.test.js [hunk 1]
```diff
@@ -72,6 +72,19 @@ test('shows dropdown', t => {
   t.snapshot(toJson(wrapper))
 })
 
+test('always shows dropdown', t => {
+  const { tree } = t.context
+  const wrapper = shallow(<DropdownTreeSelect id={dropdownId} data={tree} showDropdownAlways />)
+  t.snapshot(toJson(wrapper))
+})
+
+test('keeps dropdown open  for showDropdownAlways', t => {
```
Comment (mrchief) [inline src/index.test.js]: The extra space in the title can be removed.

Comment (mrchief) [reply src/index.test.js]: Other than that, this looks OK. @ellinge thoughts?

Comment (moonjy1993) [reply src/index.test.js]: @mrchief  hi, which line exactly are you referring to when you said "The extra space in the title"?

Comment (moonjy1993) [reply src/index.test.js]: yap. deleted the extra space.

Comment (ellinge) [reply src/index.test.js]: Probably could avoid some event handling etc if's always open, but I guess this is simple enough so never mind.

We have to make sure it's not closed with keyboard in some way also in the other PR, I think that it won't be a problem though. Not sure if we need to make any changes with ARIA as well but that's not part of this at this moment at least.

**Comment by mrchief** [comment]
@allcontributors[bot] Let's add @moonjy1993 for code, bug

**Pull request merged**